### PR TITLE
Add documentation on creating composites

### LINKED ISSUE #511: Add/update documentation about composites and compositors
**Is your feature request related to a problem? Please describe.**
The current documentation about has very limited information about different compositors, and how to create custom composites.

**Describe the solution you'd like**
Expand documentation about composites and how to define them in code and configuration files, and add a section on compositors.

Related issues: #472 #460 
Related pull requests: #429 #498 #503

### PR DESCRIPTION
**Pull request by pnuu**
This PR updates the documentatin on compositors and how the composites can be created and further configured in YAML.

 - [x] Closes #511

### CODE REVIEW

**Comment by djhoese** [comment]
I really think we should consider using the autodoc functionality and moving this documentation either in to the `satpy/composites/__init__.py` module or using `.. autosummary::` or similarly `.. autoclass::` for the composite classes. One difficulty is that the autoclass (and other autodoc sections) aren't meant to be duplicated so we'd have to think about how to do that best. I also noticed that autosummary wasn't giving a useful number of the arguments in the generated docs. Maybe there are options for both that could be useful.

File: doc/source/composites.rst [hunk 1]
```diff
@@ -2,48 +2,405 @@
...
+first composite will be placed on the day-side of the scene, and the
+second one on the night side.  The transition from day to night is
+done by calculating solar zenith angle (SZA) weighed average of the
+two composites.  The SZA can optionally be given as third dataset, and
+if not given, the angles will be calculated.  Width of the blending
+zone can be defined when initializing the compositor (default values
+shown in the example below).
+
+    >>> from satpy.composites import DayNightCompositor
+    >>> compositor = DayNightCompositor("dnc", lim_low=85., lim_high=95.)
```
Comment (gerritholl) [inline doc/source/composites.rst]: This fails with:

```TypeError: __init__() got multiple values for argument 'lim_low'```

`DayNightCompositor.__init__` takes no positional arguments:

https://github.com/pytroll/satpy/blob/dff69ad5dc8373961111b4e1956f83e92a68618c/satpy/composites/__init__.py#L905

File: doc/source/composites.rst [hunk 2]
```diff
@@ -2,48 +2,405 @@
...
+RealisticColors
+---------------
+
+:class:`RealisticColors` compositor is a special compositor that is
+used to create realistic near-true-color composite from MSG/SEVIRI
+data::
+
+    >>> from satpy.composites import RealisticColors
+    >>> compositor = RealisticColors("realcols", lim_low=85., lim_high=95.)
+    >>> composite = compositor([local_scene['VIS_006'],
```
Comment (gerritholl) [inline doc/source/composites.rst]: This should be `VIS006`:

https://github.com/pytroll/satpy/blob/dff69ad5dc8373961111b4e1956f83e92a68618c/satpy/etc/readers/seviri_l1b_hrit.yaml#L271

File: doc/source/composites.rst [hunk 3]
```diff
@@ -2,48 +2,405 @@
...
+    ...                         local_scene['VIS_008'],
```
Comment (gerritholl) [inline doc/source/composites.rst]: This should be `VIS008`:

https://github.com/pytroll/satpy/blob/dff69ad5dc8373961111b4e1956f83e92a68618c/satpy/etc/readers/seviri_l1b_hrit.yaml#L287

File: doc/source/composites.rst (hunk 2 above)
Comment (djhoese) [reply doc/source/composites.rst]: Is there another reader with the name `VIS_006`?

File: doc/source/composites.rst (hunk 1 above)
Comment (djhoese) [reply doc/source/composites.rst]: This compositor should probably change to having `name` as the first positional argument to match the parent class. Thoughts @pnuu ?

File: doc/source/composites.rst (hunk 2 above)
Comment (pnuu) [reply doc/source/composites.rst]: No. It's a typo I make frequently. Will fix in a bit.

File: doc/source/composites.rst [hunk 4]
```diff
@@ -2,48 +2,405 @@
...
-is defined as a special dataset which may have several bands (like R, G, B bands). However, the data isn't stretched,
-or clipped or gamma filtered until an image is generated.
+ - ``transition_min``: values below or equal to this are clouds -> opaque white
+ - ``transition_max``: values above this are cloud free -> transparent
+ - ``transition_gamma``: gamma correction applied to clarify the clouds
 
+Usage (with default values)::
+
+    >>> from satpy.composites import CloudCompositor
+    >>> compositor = CloudCompositor("clouds", transition_min=258.15,
```
Comment (gerritholl) [inline doc/source/composites.rst]: This has the same problem as `DayNightCompositor`; there is no positional argument (see [my comment there](https://github.com/pytroll/satpy/pull/513/files#r273089889)):

https://github.com/pytroll/satpy/blob/dff69ad5dc8373961111b4e1956f83e92a68618c/satpy/composites/__init__.py#L1135

so this fails again with `TypeError: __init__() got multiple values for argument 'transition_min'`.

File: doc/source/composites.rst (hunk 1 above)
Comment (pnuu) [reply doc/source/composites.rst]: I guess so, I'll update.

File: doc/source/composites.rst [hunk 5]
```diff
@@ -2,48 +2,405 @@
...
+
     >>> from satpy.writers import to_image
     >>> img = to_image(composite)
     >>> img.invert([False, False, True])
     >>> img.stretch("linear")
     >>> img.gamma(1.7)
     >>> img.show()
 
+This part is called ``enhancement``, and is covered in more detail
+`enhancements <enhancements.html>`_ section.
```
[the remark this answers is not in the document]
Comment (pnuu) [reply doc/source/composites.rst]: Added a proper `:doc:` reference.

[the remark this answers is not in the document]
Comment (djhoese) [reply doc/source/composites.rst]: Github isn't showing the `:doc:` or any of your recent commits.

[the remark this answers is not in the document]
Comment (djhoese) [reply doc/source/composites.rst]: This PR is based on a branch on the main repository I think (not @pnuu).

**Comment by pnuu** [comment]
OK, now the new commits are made to the `origin`, not in my fork. Ye olden days smiteth back.

**Comment by pnuu** [comment]
Something went pretty much wrong in some point. I'll make a new PR from a new branch in my fork with these changes tomorrow and then close this one.

**Pull request closed without merging**